### PR TITLE
modularize the commands

### PR DESCRIPTION
**Pull request by JuanPabloAcosta112**
This is the way in which we will perform the modularization of the commands, we will continue working on the migration of the plugins.

### CODE REVIEW

**Review by zhm — CHANGES_REQUESTED** [review]
I see this PR adds a `package-lock.json`. Is that intentional? If we're using `yarn` I would not expect to see that. If we're switching to npm we should delete the `yarn.lock`.

**Comment by zhm** [comment]
Changes look great 👌 . Once we get the lockfile fixed we can merge this. 😎

File: src/commands/cli.js [hunk 1]
```diff
@@ -1,30 +1,21 @@
...
-import Console from './console';
 import fulcrumPackage from '../version';
-
 import { Database } from 'minidb';
 
-yargs.$0 = 'fulcrum';
-
 require('source-map-support').install();
 
+/*let query, setup, sync, reset;
```
Comment (LeonardoCelisSpatial) [inline src/commands/cli.js:11]: remove unused comments 💅

File: src/commands/cli.js [hunk 2]
```diff
@@ -1,30 +1,21 @@
...
+
 const COMMANDS = [
-  Setup,
-  Sync,
-  Reset,
-  Query,
-  Console
-];
+  query = require('../cmds/query-module'),
```
Comment (LeonardoCelisSpatial) [inline src/commands/cli.js:14]: Remove the unused code 💅

File: src/commands/cli.js [hunk 3]
```diff
@@ -52,49 +43,46 @@ export default class CLI {
   async run() {
     let cli = this.yargs.usage('Usage: fulcrum <cmd> [args]');
 
-    cli.$0 = 'fulcrum';
-
+    //cli.$0 = 'fulcrum';
```
Comment (LeonardoCelisSpatial) [inline src/commands/cli.js:46]: Remove the unused code 💅

File: src/commands/cli.js [hunk 4]
```diff
@@ -52,49 +43,46 @@ export default class CLI {
...
     // this is some hacks to coordinate the yargs handler function with this async function.
     // if yargs adds support for promises in the command handlers this can go away.
     let promiseResolve = null;
     let promiseReject = null;
 
-    const completion = new Promise((resolve, reject) => {
+    /*const completion = new Promise((resolve, reject) => {
```
Comment (LeonardoCelisSpatial) [inline src/commands/cli.js:52]: Remove the hacky code that is no longer needed 💅

File: src/commands/cli.js [hunk 5]
```diff
@@ -52,49 +43,46 @@ export default class CLI {
...
-         .argv;
+    this.argv = await cli.commandDir('cmds')
+        .demandCommand()
+        .version(fulcrumPackage.version)
+        .scriptName('')
+        .help()
+        .argv
 
-    await completion;
+    //await completion;
```
Comment (LeonardoCelisSpatial) [inline src/commands/cli.js:85]: Remove the unused code 💅

**Pull request merged**